src/ILVerify/src/ILImporter.Verify.cs

```diff
@@ -1694,6 +1733,83 @@ void CheckPendingPrefix(Prefix mask)
             Check((mask & Prefix.Constrained) == 0, VerifierError.Constrained);
         }
 
+        bool HasPendingPrefix(Prefix prefix)
+        {
+            return (_pendingPrefix & prefix) != 0;
+        }
+
+        void ClearPendingPrefix(Prefix prefix)
+        {
+            _pendingPrefix &= ~prefix;
+        }
+
+        void VerifyPendingPrefix(ILOpcode opCode)
+        {
+            if (_pendingPrefix == 0)
+                return;
+
+            if (HasPendingPrefix(Prefix.Volatile) || HasPendingPrefix(Prefix.Unaligned))
+            {
+                switch (opCode)
+                {
+                    case ILOpcode.ldind_i:
```

Comment:
The logic on which instruction supports which prefix is kind of duplicated: the prefixes are cleared in each instruction, and we have also the list here.

Would it make sense to verify that all prefix flags have been cleared after the instruction is imported, and only report the error then? It would duplicating the list here.